homeassistant/components/gogogate2/cover.py

```diff
@@ -1,112 +1,190 @@
+
+    async_add_entities(
+        [
+            Gogogate2Cover(config_entry, data_update_coordinator, door)
+            for door in get_configured_doors(data_update_coordinator.data)
+        ]
+    )
+
+
+class Gogogate2Cover(CoverEntity):
+    """Cover entity for goggate2."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        data_update_coordinator: GogoGateDataUpdateCoordinator,
+        door: Door,
+    ) -> None:
+        """Initialize the object."""
+        self._config_entry = config_entry
+        self._data_update_coordinator = data_update_coordinator
+        self._door = door
+        self._api = data_update_coordinator.api
+        self._unique_id = cover_unique_id(config_entry, door)
+        self._is_available = True
+        self._transition_state: Optional[str] = None
+        self._transition_state_start: Optional[datetime] = None
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._is_available
 
-class MyGogogate2Device(CoverEntity):
-    """Representation of a Gogogate2 cover."""
+    @property
+    def should_poll(self) -> bool:
+        """Return False as the data manager handles dispatching data."""
+        return False
 
-    def __init__(self, mygogogate2, device, name):
-        """Initialize with API object, device id."""
-        self.mygogogate2 = mygogogate2
-        self.device_id = device["door"]
-        self._name = name or device["name"]
-        self._status = device["status"]
-        self._available = None
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return self._unique_id
 
     @property
     def name(self):
-        """Return the name of the garage door if any."""
-        return self._name if self._name else DEFAULT_NAME
+        """Return the name of the door."""
+        return self._door.name
 
     @property
     def is_closed(self):
         """Return true if cover is closed, else False."""
-        return self._status == STATE_CLOSED
+        if self._door.status == DoorStatus.OPENED:
+            return False
+        if self._door.status == DoorStatus.CLOSED:
+            return True
+
+        return None
+
+    @property
+    def is_opening(self):
+        """Return if the cover is opening or not."""
+        return self._transition_state == STATE_OPENING
+
+    @property
+    def is_closing(self):
+        """Return if the cover is closing or not."""
+        return self._transition_state == STATE_CLOSING
 
     @property
     def device_class(self):
         """Return the class of this device, from component DEVICE_CLASSES."""
-        return "garage"
+        return DEVICE_CLASS_GARAGE
 
     @property
     def supported_features(self):
         """Flag supported features."""
         return SUPPORT_OPEN | SUPPORT_CLOSE
 
+    async def async_open_cover(self, **kwargs):
```

Comment:
Why use the async api when the device library doesn't support that? It doesn't seem needed.